source/DSCResources/DSC_PostgreSqlScript/DSC_PostgreSqlScript.psm1

```diff
@@ -0,0 +1,268 @@
+        GetFilePath      = [System.String] $GetFilePath
+        TestFilePath     = [System.String] $TestFilePath
+        GetResult        = [System.String[]] $getResult
+    }
+
+    return $returnValue
+}
+
+
+<#
+    .SYNOPSIS
+        Executes the set-script.
+    .PARAMETER DatabaseName
+        Specifies the name of the PostgreSQL database.
+    .PARAMETER SetFilePath
+        Path to the T-SQL file that will perform Set action.
+    .PARAMETER GetFilePath
+        Path to the T-SQL file that will perform Get action.
+        Any values returned by the T-SQL queries will also be returned by the cmdlet Get-DscConfiguration through the `GetResult` property.
+    .PARAMETER TestFilePath
+        Path to the T-SQL file that will perform Test action.
+        Any script that does not throw an error or returns null is evaluated to true.
+        The cmdlet Invoke-Sqlcmd treats T-SQL Print statements as verbose text, and will not cause the test to return false.
+    .PARAMETER Credential
+        The credentials to authenticate with, using PostgreSQL Authentication.
+    .PARAMETER PsqlLocation
+        Location of the psql executable.  Defaults to "C:\Program Files\PostgreSQL\12\bin\psql.exe".
+    .PARAMETER CreateDatabase
+        Optionally creates a database if the database specified with DatabaseName doesn't exist.  Defaults to $true.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $DatabaseName,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $SetFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $GetFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TestFilePath,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $PsqlLocation = "C:\Program Files\PostgreSQL\12\bin\psql.exe",
+
+        [Parameter()]
+        [bool]
+        $CreateDatabase = $true
+    )
+
+    $env:PGPASSWORD = $Credential.Password.ToString()
+    $env:PGUSER = $Credential.UserName
+
+    try
+    {
+        Write-Verbose -Message ($script:localizedData.ExecutingSetScript -f $SetFilePath,$DatabaseName)
+        $previousErrorActionPreference = $ErrorActionPreference
+        $ErrorActionPreference = "Stop"
+        $DbExists = $false;
+        $DbList = Invoke-Command -ScriptBlock {& $PsqlLocation -lqt 2>&1}
+
+        foreach ($db in $DbList)
+        {
+            if ($db.split("|")[0].trim() -eq $DatabaseName)
+            {
+                $DbExists = $true
+                continue
+            }
+        }
+
+        if ($DbExists -eq $false -and $CreateDatabase)
+        {
+            Invoke-Command -ScriptBlock {
+                & $PsqlLocation -c "CREATE DATABASE $DatabaseName"
```

Comment:
I am not sure how I feel about including this in the same resource. Typically I would split these since checking for the database isn't specifically tested for.